pkg/projects/resource_project.go

```diff
@@ -177,18 +177,58 @@ func projectResource() *schema.Resource {
 			},
 			Description: "Project group. Element has one to one mapping with the [JFrog Project Groups API](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-UpdateGroupinProject)",
 		},
+
+		"role": {
+			Type:     schema.TypeSet,
+			Optional: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"name": {
+						Type:     schema.TypeString,
+						Required: true,
+						ValidateDiagFunc: validation.ToDiagFunc(validation.All(
+							validation.StringIsNotEmpty,
+							maxLength(64),
+						)),
+					},
+					"description": {
+						Type:     schema.TypeString,
+						Optional: true,
+					},
+					"type": {
+						Type:             schema.TypeString,
+						Required:         true,
+						ValidateDiagFunc: validation.ToDiagFunc(validation.StringMatch(customRoleTypeRegex, fmt.Sprintf("Only \"%s\" is supported", customRoleType))),
```

Comment:
in the future, if you have `""` in the string you can use the  gravemark string and then you don't need to escape

Reply:
Yeah, will update. Interesting you call it gravemark, I always know it as backtick.